### PR TITLE
feat: optimsitic assertor happy path test and timer integration

### PR DESCRIPTION
**Pull request by md0x**
Signed-off-by: Pablo Maldonado <pablo@umaproject.org>

**Summary**

- Add first lifecycle assertion test
- Add `OptimisticAssertorTest` contract to override time
- Add `getCurrentTime` function

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [x]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

Fixes #XXXX

### CODE REVIEW

File: packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol [hunk 1]
```diff
@@ -155,13 +155,13 @@ contract OptimisticAssertor is Lockable, OptimisticAssertorInterface, Ownable {
     }
 
     function settleAssertion(bytes32 assertionId) public {
-        Assertion memory assertion = assertions[assertionId];
+        Assertion storage assertion = assertions[assertionId];
```
Comment (Reinis-FRP) [inline packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol:158]: good catch :)

File: packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    function testAssertionWithNoDispute() public {
+        vm.startPrank(TestAddress.account1);
+        defaultCurrency.allocateTo(TestAddress.account1, optimisticAssertor.defaultBond());
+        assert(defaultCurrency.balanceOf(TestAddress.account1) >= optimisticAssertor.defaultBond());
+        defaultCurrency.approve(address(optimisticAssertor), optimisticAssertor.defaultBond());
+
+        bytes32 assertionId = optimisticAssertor.assertTruth(bytes(claimAssertion));
+
+        // Move time forward to the end of the liveness period.
```
Comment (Reinis-FRP) [inline packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol:29]: before time travel we should also check that `settleAndGetAssertion` would revert

File: packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        timer.setCurrentTime(timer.getCurrentTime() + optimisticAssertor.defaultLiveness());
+
+        // The assertion should be true.
+        assert(optimisticAssertor.settleAndGetAssertion(assertionId));
```
Comment (Reinis-FRP) [inline packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol:33]: as we are using `assertTruth` that currently passes `address(0)` as `proposer` I wonder why this test did not fail? internal `settleAssertion` should have reverted due to non-existing `proposer` address!

File: packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol (hunk 2 above)
Comment (Reinis-FRP) [reply packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol:29]: also catching the reason "Assertion not expired"

File: packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        timer = oaContracts.timer;
+    }
+
+    function testAssertionWithNoDispute() public {
+        vm.startPrank(TestAddress.account1);
+        defaultCurrency.allocateTo(TestAddress.account1, optimisticAssertor.defaultBond());
+        assert(defaultCurrency.balanceOf(TestAddress.account1) >= optimisticAssertor.defaultBond());
+        defaultCurrency.approve(address(optimisticAssertor), optimisticAssertor.defaultBond());
+
+        bytes32 assertionId = optimisticAssertor.assertTruth(bytes(claimAssertion));
```
Comment (Reinis-FRP) [inline packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol:27]: could also check balances that bond was transferred.

File: packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol (hunk 3 above)
Comment (Reinis-FRP) [reply packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol:33]: nevermind, my mistake as internal `assertTruthFor` by default passes msg.sender

File: packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol (hunk 2 above)
Comment (md0x) [reply packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol:42]: Good points, added in latest commit

File: packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                bytes(claimAssertion),
+                TestAddress.account1,
+                address(0),
+                address(0),
+                defaultCurrency,
+                optimisticAssertor.defaultBond(),
+                optimisticAssertor.defaultLiveness()
+            );
+
+        // Settle before the liveness period shoul revert.
```
Comment (md0x) [inline packages/core/test/foundry/optimistic-assertor/OptimisticAsserter.Lifecycle.t.sol:38]:
```suggestion
        // Settle before the liveness period should revert.
```

File: packages/core/test/foundry/fixtures/optimistic-assertor/OptimisticAssertorFixture.sol [hunk 6]
```diff
@@ -29,8 +29,14 @@ contract OptimisticAssertorFixture is Test {
         baseMockDvmContracts.identifierWhitelist.addSupportedIdentifier("ASSERT_TRUTH");
```
[the remark this answers is not in the document]
Comment (md0x) [reply packages/core/test/foundry/fixtures/optimistic-assertor/OptimisticAssertorFixture.sol:29]: Added a separate test for this

**Review by Reinis-FRP — APPROVED** [review]
looks good!

**Pull request merged**